pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -67,49 +67,59 @@ This file implements string parsing and creation for NumPy datetime.
  * Returns 0 on success, -1 on failure.
  */
 
+enum Comparison {
```

Comment:
Design wise this assumes that the callee knows what the caller is doing and can instruct it on actions to take. I think it would be better to separate those entities and just have the callee report back what it knows.

With that in mind, maybe call rename this to `DatetimePartParseResult` and maybe have values of `PARTIAL_MATCH`, `EXACT_MATCH`, `NO_MATCH`. The caller can then choose to take action independent of this function

Reply:
As in, to name the values the same way as those from `FormatRequirement`?

The issue is that different format requirements can result in the same result from this function - for example, both `EXACT_MATCH` where the format matches and `INFER_FORMAT` can return `0`

I've renamed the values to
```
    COMPARISON_SUCCESS,
    COMPLETED_PARTIAL_MATCH,
    COMPARISON_ERROR
```
, is that clearer?